### PR TITLE
Add PHP8 support to Spryker harness

### PR DESCRIPTION
**Pull request by hgajjar**

### CODE REVIEW

File: src/spryker/application/skeleton/composer-harness.json.twig [hunk 1]
```diff
@@ -59,11 +59,18 @@
...
-    "compatibility": "bin/phpcs -s --standard=vendor/phpcompatibility/php-compatibility/PHPCompatibility/ --runtime-set testVersion 7.4-"
+    "compatibility": "bin/phpcs -s --standard=vendor/phpcompatibility/php-compatibility/PHPCompatibility/ --runtime-set testVersion {{ @('php.version') }}-",
+    "phpstan": [
+      "php -d memory_limit=1024M vendor/bin/phpstan analyze -c phpstan.neon src/ -l 5"
+    ]
   },
   "config": {
     "platform": {
-      "php": "7.4.10"
+      "php": "{{ @('php.version') }}"
```
Comment (andytson-inviqa) [inline src/spryker/application/skeleton/composer-harness.json.twig]: just a note here, that if php.version is 7.4, that will likely be taken to mean 7.4.0, and composer reqs that do things like > 7.4.3 will fail or install the wrong version of code

File: src/spryker/application/overlay/phpstan-baseline.neon [hunk 2]
```diff
@@ -1,51 +1,11 @@
...
+	ignoreErrors:
+		-
+			message: "#^Parameter \\#1 \\$array of function array_filter expects array, string given\\.$#"
+			count: 1
+			path: src/Pyz/Yves/ProductSetWidget/Widget/ProductSetIdsWidget.php
+
+		-
+			message: "#^Parameter \\#2 \\$default of method Symfony\\\\Component\\\\HttpFoundation\\\\InputBag\\:\\:get\\(\\) expects bool\\|float\\|int\\|string\\|null, array given\\.$#"
+			count: 1
+			path: src/Pyz/Yves/ProductSetWidget/Widget/ProductSetIdsWidget.php
```
Comment (andytson-inviqa) [inline src/spryker/application/overlay/phpstan-baseline.neon]: As this is an overlay file, it will affect existing projects

Comment (hgajjar) [reply src/spryker/application/overlay/phpstan-baseline.neon]: It is moved to skeleton now

File: src/spryker/application/skeleton/phpstan.neon [hunk 3]
```diff
@@ -6,6 +6,7 @@ parameters:
     excludes_analyse:
         - %rootDir%/../../../src/Generated/*
         - %rootDir%/../../../src/Orm/*
+        - %rootDir%/../../../src/Pyz/Zed/DataImport/Business/*
```
Comment (hgajjar) [inline src/spryker/application/skeleton/phpstan.neon]: This module has some phpstan issues, which I couldn't fix easily. That is why I have ignored it for now.

**Comment by hgajjar** [comment]
@andytson-inviqa It's ready for review now

**Review by kierenevans — DISMISSED** [review]

**Review by andytson-inviqa — DISMISSED** [review]

File: src/spryker/application/skeleton/composer-harness.json.twig [hunk 4]
```diff
@@ -63,21 +86,32 @@
...
       "dealerdirect/phpcodesniffer-composer-installer": true,
       "sllh/composer-versions-check": true
     },
     "platform": {
-      "php": "7.4.10"
+      "php": "{{ version_compare(@('spryker.demoshop-version'), '202108.0', '<') ? '7.4.10' : '8.0.16' }}"
+    },
+    "allow-plugins": {
+      "dealerdirect/phpcodesniffer-composer-installer": true,
+      "sllh/composer-versions-check": true
```
Comment (andytson-inviqa) [inline src/spryker/application/skeleton/composer-harness.json.twig]:
```suggestion
```

**Review by andytson-inviqa — APPROVED** [review]

**Pull request merged**